sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlockBlobAsyncClient.java

```diff
@@ -0,0 +1,346 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.specialized.cryptography;
+
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.cryptography.AsyncKeyEncryptionKey;
+import com.azure.core.http.rest.Response;
+import com.azure.core.implementation.util.FluxUtil;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.security.keyvault.keys.cryptography.models.KeyWrapAlgorithm;
+import com.azure.storage.blob.BlobAsyncClient;
+import com.azure.storage.blob.implementation.AzureBlobStorageImpl;
+import com.azure.storage.blob.models.AccessTier;
+import com.azure.storage.blob.models.BlobAccessConditions;
+import com.azure.storage.blob.models.BlobHTTPHeaders;
+import com.azure.storage.blob.models.BlockBlobItem;
+import com.azure.storage.blob.models.ParallelTransferOptions;
+import com.azure.storage.blob.specialized.BlockBlobAsyncClient;
+import com.azure.storage.common.Constants;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import javax.crypto.Cipher;
+import javax.crypto.KeyGenerator;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.ShortBufferException;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.AsynchronousFileChannel;
+import java.security.GeneralSecurityException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * This class provides a client side encryption client that contains generic blob operations for Azure Storage Blobs.
+ * Operations allowed by the client are uploading, downloading and copying a blob, retrieving and setting metadata,
+ * retrieving and setting HTTP headers, and deleting and un-deleting a blob. The upload and download operation allow for
+ * encryption and decryption of the data client side.
```

Comment:
> side. [](start = 47, length = 6)

Can we add a call out that setting metadata in particular is unsafe and should only be done with caution?